### ISSUE TITLE
Add option to specify portal culling properties for terrain meshes and instancer meshes

### ISSUE THREAD

**Issue by m4nu3lf** [Zylann/godot_voxel#343]
I'm currently working on a project which uses the plugin to generate an asteroid terrain.
I'm also planning to use Godot's built-in room/portal system for internal structures.

Creating a mesh instance in the visual server with culling mode "Static" will cause the mesh not to be visible if there is a room system active. Because of this, The terrain won't be visible when generating until the room data is updated (which you can't do every time the terrain change).

I modified the DirectMeshInstance code to set the terrain meshes to ROAMING mode.
Both GLOBAL and ROAMING modes should allow the terrain to work when a room system is active. The latter enables terrain meshes to be culled by portals! (However, I don't know how this would perform for large terrains with lots of meshes).

Either way, it would be good to expose the portal culling properties for the terrain meshes in the editor.
I have hardcoded the values I need in a patch I'm going to use for my project, but I assume other people would want to use this plugin together with the room system.

**Comment by Zylann** [comment]
Where are these properties usually on a MeshInstance? I can't seem to find it in the doc

**Comment by m4nu3lf** [comment]
> ~Where are these properties usually on a MeshInstance? I can't seem to find it in the doc~ Oh, CullInstance... almost forgot that the entire branch of inheritance had this inserted just so room and portals could work
> 
> I'm not familiar with this system, but also the last Godot3 version of this module has been `godot3.4`, which is a feature freeze. There is a bit of extra in `master` (with some bugs) but I had planned to stop working on that version to focus on Godot4 from now on, which would later be merged as the master branch. I can add that in the `godot4` branch, but adding it to a Godot3-compatible version has lower priority now, and will need backporting (as the commit will not be compatible).

Most options of the CullInstance are only relevant for STATIC/DYNAMIC instances (which must have the same lifecycle of the room system). So they do not apply to the Voxel Terrain. Thinking about it even the ROAMING mode can cause issues as it could assign big meshes to small rooms if their AABB has a centre within the room.
I think the best solution would be to just set both DIrectMeshInstance and DirectMultimeshInstance to GLOBAL when they are constructed with the VisualServer.
In this case, I would consider it more like an (easy) bug fix rather than a new functionality, as right now the terrain won't be visible if there is a room system active.

**Comment by slapin** [comment]
@m4nu3lf do you have your local patch I could use as it seems Zylann stopped support for 3.x version so now we're out of luck because Godot4 is broken like brick and Godot3.x is no longer supported :(

**Comment by m4nu3lf** [comment]
> @m4nu3lf do you have your local patch I could use as it seems Zylann stopped support for 3.x version so now we're out of luck because Godot4 is broken like brick and Godot3.x is no longer supported :(

Yes, I'll create a PR. @Zylann  is master still meant to work with Godot 3.x?

**Comment by Zylann** [comment]
>  is master still meant to work with Godot 3.x?

No, master targets Godot 4.0.

**Comment by m4nu3lf** [comment]
@Zylann @slapin  I created this one https://github.com/Zylann/godot_voxel/pull/351
